### ISSUE TITLE
Mautic Segments are not populating when using filter based on select field 

### ISSUE THREAD

**Issue by Berlira** [mautic/mautic#3866]
What type of report is this:

BUG REPORT 

| Bug report? | 
| Feature request? | 
| Enhancement? | 

## Description:
Hello - I have a custom field and when I filter for it in Mautic segments, segments are no longer populating with contacts.  I just get "no contacts".  

I checked the logs and this is what is says and dropped it down below. 

| Mautic version |  2.7.1
| PHP version | 5.6.29

### Log errors: 

mautic.NOTICE: Doctrine\DBAL\Exception\SyntaxErrorException: An exception occurred while executing 'SELECT count(l.id) as lead_count, max(l.id) as max_id FROM ma_leads l LEFT JOIN ma_lead_lists_leads ll ON (ll.leadlist_id = 7) AND (ll.lead_id = l.id) AND (ll.date_added <= '2017-04-17 17:57:02') WHERE (()) AND (ll.lead_id IS NULL)':  SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ')) AND (ll.lead_id IS NULL)' at line 1 (uncaught exception) at /opt/bitnami/apps/mautic/htdocs/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 90 while running console command `mautic:segments:update` [] []

**Comment by Berlira** [comment]
Oh....also - I originally had this field under Contacts/core, but moved it to Contacts/Professional - wasn't working and moved it back to Core in the hopes it would work again, but nothing. 

Not sure if this had any effect.

**Comment by escopecz** [comment]
> I originally had this field...

What field? Please provide steps to replicate the issue. All of your segments stopped working or only some specific? Please provide information which will help developers to replicate it and fix it.

**Comment by Berlira** [comment]
Hi - 

Sure.  I had a custom select (drop-down) field with 8 values 

*	New Contact 
*	Tour Scheduled 
*	Tour Completed 
*	Enrolled 
*	Closed – Clost 
*	Closed – Dropped
*	Closed Bad 
*	Wait Listed 

 

When I use this field as a filter in any segment, it does not pull leads that qualify, even if it is the only qualifier I am using in the segment.  

 

So I reduced the number of values to 6 values -  New, Tour Scheduled, Tour Completed, Enrolled, Closed, Wait Listed – But I continue to have the same issue. 

 

However, I am noticing that the segments do not pull any leads at all when the field is a ‘select’ field.  So even if the field is a Mautic field (i.e. Stage) I still have the same issue.  

 

I checked the log and got the syntax error message I mentioned when I first submitted the ticket.  

 

Any help would be much appreciated.  Thanks.
